### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by sullenor** [npm/npm#9651]
Hi, I tried the new version of npm today and ended up with an error 'Maximum call stack size exceeded' during installing project dependencies.

```bash
npm ERR! Darwin 14.5.0
npm ERR! argv "node" "/usr/local/bin/npm" "i"
npm ERR! node v0.12.0
npm ERR! npm  v3.3.3

npm ERR! Maximum call stack size exceeded
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
npm ERR! Darwin 14.5.0
npm ERR! argv "node" "/usr/local/bin/npm" "i"
npm ERR! node v0.12.0
npm ERR! npm  v3.3.3

npm ERR! Maximum call stack size exceeded
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
```

You may see the whole `npm-debug.log` file by url: https://yadi.sk/i/9-pg2asOjAgii

**Comment by llaughlin** [comment]
Same here: https://gist.github.com/llaughlin/84fb5866b0735e63bdbc

**Comment by othiym23** [comment]
Closing as a duplicate of #9224, which is the tracking issue for this problem. We've had a tough time reproducing this issue, so if it's reproducibly crashing up for this application, and you can share your `package.json`, it would be a big help to us.

To get yourself unblocked, you can try

1. upgrading to a newer version of node, because 0.12.0 has a number of pretty major issues that got addressed in subsequent releases of Node 0.12, and
2. `rm -r node_modules && npm install` to see if the problem is a weird configuration of modules in the existing tree

Good luck, and thanks for the debug log! Those are very useful as we try to figure out what's going on here!

**Comment by sullenor** [comment]
Sorry, haven't found this one.
Unfortunately, we use a lot of private packages for our project and I can't share them. But, I'll try to reproduce it with public ones.

Thanks

**Comment by JESii** [comment]
Here's a failure; adding npm-debug.log. No package.json as I just want to install Appium. here's the command line output.
```
yc98js1@a4660809:[dev->]~/dev$ npm install wd
npm ERR! Darwin 15.3.0
npm ERR! argv "/usr/local/Cellar/node/5.9.0/bin/node" "/usr/local/bin/npm" "install" "wd"
npm ERR! node v5.9.0
npm ERR! npm  v3.7.3

npm ERR! Maximum call stack size exceeded
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/yc98js1/dev/npm-debug.log
```
[npm-debug.txt](https://github.com/npm/npm/files/179957/npm-debug.txt)

**Comment by chaseadamsio** [comment]
On a Centos 7.0 virtual machine:

```
14:42:31 npm ERR! Maximum call stack size exceeded
14:42:31 npm ERR! 
14:42:31 npm ERR! If you need help, you may report this error at:
14:42:31 npm ERR!     <https://github.com/npm/npm/issues>
14:42:31 npm ERR! Linux 3.10.0-123.el7.x86_64
14:42:31 npm ERR! argv "/app/.nvm/versions/node/v4.2.4/bin/node" "/app/.nvm/versions/node/v4.2.4/bin/npm" "install"
14:42:31 npm ERR! node v4.2.4
14:42:31 npm ERR! npm  v3.6.0
```

**Comment by JESii** [comment]
@chaseadamsio - I just created a new issue since this one is closed; see https://github.com/npm/npm/issues/12047
